src/Stratis.Bitcoin.Features.Api/ApiSettings.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using Stratis.Bitcoin.Configuration;
+
+namespace Stratis.Bitcoin.Features.Api
+{
+    /// <summary>
+    /// Configuration related to storage of transactions.
+    /// </summary>
+    public class ApiSettings
+    {
+        /// <summary>URI to node's API interface.</summary>
+        public Uri ApiUri { get; set; }       
+
+        private Action<ApiSettings> callback = null;
```

Comment:
I'm not sure if you need this or not. I haven't really been able to figure out what exactly the usage for this is. I know when I moved ConsensusSettings into it's own class I didn't bring this callback stuff along and it has had no ill effects. I recommend if you have no code consuming then get rid of it.

Reply:
The callback provides a way to provide startup options - e.g. `.UseApi(s => s.ApiUri = "http://someuri")`

Reply:
Ah yes that's kind of cool. I wonder if I should also add that feature to consensus settings as it currently doesn't have that support.